Adapter.C

```diff
@@ -792,100 +789,60 @@ void preciceAdapter::Adapter::reloadMeshPoints()
         return;
     }
 
-    // In Foam::polyMesh::movePoints.
-    // TODO: The function movePoints overwrites the pointer to the old mesh.
-    // Therefore, if you revert the mesh, the oldpointer will be set to the points, which are the new values.
-    DEBUG(adapterInfo("Moving mesh points to their previous locations..."));
+    // Invalidate on-demand fields
+    // fvMesh::updateGeomNotOldVol()
+    // (void)mesh_.V();
+    (void)mesh_.Sf();
+    (void)mesh_.magSf();
+    (void)mesh_.C();
+    (void)mesh_.Cf();
 
-    // TODO
-    // Switch oldpoints on for pure physics. (is this required?). Switch off for better mesh deformation capabilities?
-    // const_cast<pointField&>(mesh_.points()) = oldMeshPoints_;
-    const_cast<fvMesh&>(mesh_).movePoints(meshPoints_);
+    // fvMesh.movePoints overwrites the pointer to the oldPoints
+    const_cast<pointField&>(mesh_.points()) = *meshOldPoints_;
 
-    DEBUG(adapterInfo("Moved mesh points to their previous locations."));
+    // Reload mesh points
+    const_cast<Foam::fvMesh&>(mesh_).movePoints(*meshPoints_);
 
-    // TODO The if statement can be removed in this case, but it is still included for clarity
-    if (meshCheckPointed)
-    {
-        readMeshCheckpoint();
-    }
+    readMeshCheckpoint();
 
-    /*  // TODO This part should only be used when sybcycling. See the description in 'storeMeshPoints()'
-        // The if statement can be removed in this case, but it is still included for clarity
-    if ( oldVolsStored )
-    {
-        readVolCheckpoint();
-    }
-    */
+    DEBUG(adapterInfo("Moved mesh points to their previous locations."));
+
+    readMeshVolCheckpoint();
```

Comment:
We need a similar comment for the volumes, or to move the message after both operations and make it for both.